### PR TITLE
feat: Timeouts for Remote Authorizer

### PR DESCRIPTION
**Pull request by kolotaev**
Remote JSON Authorizer uses Resilient HTTP Client with hard-coded minimal timeout.
While 500ms is generally considered recommended latency for such systems, latencies for real-world scenarios and setups may vary, thus it'd be good to allow configuration of Resilient HTTP Client to tailor its behavior to particular Authorizer latencies.

This PR is kind of POC (but also is a real working code) that fixes only Remote JSON Authorizer. The issue in question can be seen in other authorizers and authenticators. Unfortunately we don't have expertise and architecture-view in Oathkeeper to propose a more generic solution.

### CODE REVIEW

**Comment by aeneasr** [comment]
Thank you, this looks great! Could you please sign the CLA (just two clicks), then we can merge it right away!

**Comment by aeneasr** [comment]
Are you still up for the changes? :) If you need any help, let us know!

**Comment by kolotaev** [comment]
Hi, sorry, I was busy with other stuff.
I've signed the CLA and updated the PR title for semantic PR conformance.
There's a "test" flow still pending, do I need to re-push the changes so that it's triggered?

**Comment by aeneasr** [comment]
Ah yeah no the problem here is that GitHub requires maintainers now to approve CI runs from people who have not contributed before...

File: pipeline/authz/remote_json.go [hunk 1]
```diff
@@ -40,11 +49,23 @@ type AuthorizerRemoteJSON struct {
...
 func NewAuthorizerRemoteJSON(c configuration.Provider) *AuthorizerRemoteJSON {
-	return &AuthorizerRemoteJSON{
+	authorizer := &AuthorizerRemoteJSON{
 		c:      c,
 		client: httpx.NewResilientClientLatencyToleranceSmall(nil),
 		t:      x.NewTemplate("remote_json"),
 	}
+	// Add configurable HTTP client if set in config
+	conf, _ := authorizer.Config(nil)
+	if conf != nil && conf.ClientConfig.Timeout > 0 && conf.ClientConfig.StopAfter > 0 {
```
[the remark this answers is not in the document]
Comment (kolotaev) [reply pipeline/authz/remote_json.go:59]: Yeah, sure! It makes sense. I need some more time to rework the solution. I'll ping you once ready. Thanks for this finding.

**Comment by aeneasr** [comment]
Marking as draft!

**Pull request closed without merging**